Add recommendations

Mellerh opened the pull request:
PR на добавление фичи Add recommendations

Mellerh committed:
add-recommendations
работа над фичей. 1.0

Mellerh committed:
add-recommendations
работа над фичей. 1.1. Пройдены тесты

Mellerh committed:
add-recommendations
работа над фичей. 1.2. Пройдены тесты и исправлены зависимости в pom.xml

Mellerh committed:
add-recommendations
работа над фичей. 1.3. Вернул url: jdbc:h2:mem:./db/filmorate;MODE=PostgreSQL; с file на mem

Ivan-Vorobev commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -182,6 +184,12 @@ public Collection<Film> findTopPopular(final Long topCount) {
     }
 
+    public Collection<Film> getRecommendations(Long userId) {
+        return filmStorage.getUserRecommendations(userId).stream()
```
здесь нужно дернуть метод сервиса `UserService.findUser` для валидации пользователя, он выбросит соответствующую ошибку если пользователя нет . По идее это должно быть частью постман запросов

Mellerh replied:
Добавил метод сервиса UserService.findUser для валидации пользователя в FilmService.getRecommendations

Ivan-Vorobev replied:
Посмотрел исправления, фишкак как раз использовать сервис, чтобы логика проверки была в едином месте. Сам метод поиска пользователя ее уже содержит и тем самым не нужно повторять проверку размазывая домен по другим методам. Так что проверка на null и выбрасывание исключения нужно убрать

Mellerh replied:
Поправил

Mellerh replied:
Поправил ещё раз)

Mellerh committed:
add-recommendations
работа над фичей. 1.4. Добавлен метод сервиса UserService.findUser для валидации пользователя в FilmService.getRecommendations

Mellerh committed:
add-recommendations
работа над фичей. 1.5. Убрал метод сервиса UserService.findUser для валидации пользователя в FilmService.getRecommendations

Mellerh committed:
add-recommendations
работа над фичей. 1.6. Скорректировал вызов метода UserService.findUser для валидации пользователя в FilmService.getRecommendations
